src/common/polyfills/userscript.ts

```diff
@@ -80,34 +80,44 @@ async function getStreamText(stream: ReadableStream) {
     return str
 }
 
-async function handleReadyStateChange(isStream: boolean, r: Tampermonkey.Response<any>) {
-    Object.assign(r, { status: r.status })
-
-    if (r.readyState === XMLHttpRequest.HEADERS_RECEIVED) {
-        if (r.status === 200 && isStream) {
-            Object.assign(r, { body: r.response })
-            return r
-        } else {
-            const respText = await getStreamText(r.response)
-            Object.assign(r, { json: () => JSON.parse(respText), text: () => respText })
-            return r
-        }
-    }
-}
-
-export async function userscriptFetch(
-    url: string,
-    { body, headers, method, signal }: RequestInit,
-    isStream = true
-): Promise<any> {
+export async function userscriptFetch(url: string, { body, headers, method, signal }: RequestInit): Promise<any> {
+    // Only Tampermonkey support streaming responses
+    // eslint-disable-next-line camelcase
+    const responseType: any = GM_info.scriptHandler === 'Tampermonkey' ? 'stream' : 'text'
```

Comment:
Would it be possible that some of the other script managers do support this? Is it feasible to use methods such as try-catches rather than hardcoded ones to determine compatibility? :thinking:

Reply:
I just remembered that there is a `GM_xmlhttpRequest.RESPONSE_TYPE_STREAM` https://github.com/Tampermonkey/tampermonkey/issues/1278#issuecomment-884363078 is available here,  but seems that only Tampermonkey has implemented it.
will use this first.